test/E2ETests/Smoke.E2E.Tests/DataRowTests.cs

```diff
@@ -23,23 +23,102 @@ public void ExecuteOnlyDerivedClassDataRowsWhenBothBaseAndDerviedClassHasDataRow
                 "DataRowTestMethod (DerivedString1)",
                 "DataRowTestMethod (DerivedString2)");
 
-            // 4 tests of BaseClass - 3 datarow result and 1 parent result
-            // 3 tests of DerivedClass - 2 datarow result and 1 parent result
+            // 4 tests of BaseClass.DataRowTestMethod - 3 datarow result and 1 parent result
+            // 3 tests of DerivedClass.DataRowTestMethod - 2 datarow result and 1 parent result
             // Total 7 tests - Making sure that DerivedClass doesn't run BaseClass tests
             this.ValidatePassedTestsCount(7);
         }
 
         [TestMethod]
-        public void ExecuteOnlyDerivedClassDataRowsWhenItOverridesBaseClassDataRows()
+        public void ExecuteOnlyDerivedClassDataRowsWhenItOverridesBaseClassDataRows_SimpleDataRows()
         {
-            this.InvokeVsTestForExecution(new string[] { TestAssembly }, testCaseFilter: "FullyQualifiedName~DerivedClass");
+            this.InvokeVsTestForExecution(new string[] { TestAssembly }, testCaseFilter: "FullyQualifiedName~DerivedClass&TestCategory~DataRowSimple");
 
             this.ValidatePassedTestsContain(
                 "DataRowTestMethod (DerivedString1)",
                 "DataRowTestMethod (DerivedString2)");
 
-            // 3 tests of DerivedClass - 2 datarow result and 1 parent result
+            // 3 tests of DerivedClass.DataRowTestMethod - 2 datarow result and 1 parent result
             this.ValidatePassedTestsCount(3);
         }
+
+        [TestMethod]
+        public void DataRowsExecuteWithRequiredAndOptionalParameters()
+        {
+            this.InvokeVsTestForExecution(new string[] { TestAssembly }, testCaseFilter: "TestCategory~DataRowSomeOptional");
+
+            this.ValidatePassedTestsContain(
+                "DataRowTestMethodWithSomeOptionalParameters (42)",
+                "DataRowTestMethodWithSomeOptionalParameters (42,BaseOptionalString1)",
+                "DataRowTestMethodWithSomeOptionalParameters (42,BaseOptionalString2,BaseOptionalString3)",
+                "DataRowTestMethodWithSomeOptionalParameters (42,BaseOptionalString4,BaseOptionalString5)",
+                "DataRowTestMethodWithSomeOptionalParameters (123)",
+                "DataRowTestMethodWithSomeOptionalParameters (123,DerivedOptionalString1)",
+                "DataRowTestMethodWithSomeOptionalParameters (123,DerivedOptionalString2,DerivedOptionalString3)");
+
+            // 5 tests of BaseClass.DataRowTestMethodWithSomeOptionalParameters - 4 datarow result and 1 parent result
+            // 4 tests of DerivedClass.DataRowTestMethodWithSomeOptionalParameters - 3 datarow result and 1 parent result
```

Comment:
@ssa3512 : We can simply these tests even more by just keeping the `DataRowTestMethodWithSomeOptionalParameters` in the base class only.

Apologies, I should have been more clear.